src/hw/msi.cpp

```diff
@@ -113,8 +115,28 @@ namespace hw
     return pcibar.start + capbar_off;
   }
 
-  msix_t::msix_t(PCI_Device& device, uint32_t cap)
-    : dev(device)
+  msix_t::msix_t(PCI_Device& device, bool dev_msix, uint32_t cap)
```

Comment:
Why do we not have a MSI alongside MSIX instead of wrapping the two together ?

Reply:
We can move it out after it works, they do have some likeness